eclipse-collections/src/main/java/org/eclipse/collections/impl/map/ordered/mutable/UnmodifiableMutableOrderedMap.java

```diff
@@ -149,9 +150,9 @@ public void clear()
     }
 
     @Override
-    public Set<K> keySet()
+    public UnmodifiableMutableSet<K> keySet()
     {
-        return Collections.unmodifiableSet(this.delegate.keySet());
+        return UnmodifiableMutableSet.of(this.delegate.keySet());
```

Comment:
@johnjstuart you should be able to write `this.delegate.keySet().asUnmodifiable()` I believe here

Reply:
@johnjstuart isn't `delegate.keySet` now returning `MutableMap` because of your changes? `delegate` is a `MutableOrderedMap`